ssl.go

```diff
@@ -0,0 +1,76 @@
+package libcrypto
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"math/big"
+	"net"
+	"os"
+	"time"
+)
+
+// GenerateCertsForHost generates a self-signed certificate for host and saves them at certPath and keyPath
+func GenerateCertsForHost(host, certPath, keyPath string) error {
```

Comment:
I think we'll want to introduce more than the `host` parameter here, we should provide a combination of `ipaddr` and `hostname` as this is a generic library and it should be able to handle more than the localhost use case.